### PR TITLE
add iron session to trpc-api generator

### PR DESCRIPTION
**Pull request by tremarkley**

### CODE REVIEW

**Comment by tremarkley** [comment]
* **#154** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/154?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tremarkley and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/api-plugin/src/generators/trpc-api-generator/files/src/constants/session.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { type IronSession, type SessionOptions } from 'iron-session'
+
+import { envVars } from './envVars'
+
+export const sessionOptions: SessionOptions = {
+  password: envVars.IRON_SESSION_SECRET,
+  cookieName: 'auth.session-token',
```
Comment (jakim929) [inline packages/api-plugin/src/generators/trpc-api-generator/files/src/constants/session.ts:7]: nit: prob wanna turn this into an envvar so we can namespace by apps?

**Review by jakim929 — APPROVED** [review]
lgtm!

File: packages/api-plugin/src/generators/trpc-api-generator/files/src/constants/session.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { type IronSession, type SessionOptions } from 'iron-session'
+
+import { envVars } from './envVars'
+
+export const sessionOptions: SessionOptions = {
+  password: envVars.IRON_SESSION_SECRET,
+  cookieName: '<%= name %>.auth.session-token',
```
Comment (jakim929) [inline packages/api-plugin/src/generators/trpc-api-generator/files/src/constants/session.ts:7]: oh cool didn't know u could do this in the generator

**Pull request merged**